src/topology/homotopy/induced_maps.lean

```diff
@@ -0,0 +1,184 @@
+
+/-- The path 0 ⟶ 1 in ulift I -/
+def upath01 : path (ulift.up 0 : ulift.{u} I) (ulift.up 1) :=
+{ to_fun := ulift.up, source' := rfl, target' := rfl }
+
+local attribute [instance] path.homotopic.setoid
+/-- The homotopy path class of 0 → 1 in `ulift I` -/
+def uhpath01 := @fundamental_groupoid.from_path (Top.of $ ulift.{u} I) _ _ ⟦upath01⟧
+
+end unit_interval
+
+namespace continuous_map.homotopy
+open fundamental_groupoid
+open category_theory
+open unit_interval (uhpath01)
+open fundamental_groupoid_functor
+
+open_locale fundamental_groupoid
+open_locale unit_interval
+
+local attribute [instance] path.homotopic.setoid
+
+
+/- We let `X` and `Y` be spaces, and `f` and `g` be homotopic maps between them -/
+variables {X Y : Top.{u}} {f g : C(X, Y)} (H : continuous_map.homotopy f g)
+  {x₀ x₁ : X} (p : from_top x₀ ⟶ x₁)
+
+/-!
+These definitions set up the following diagram, for each path `p`:
+
+          f(p)
+      *--------*
+      | \      |
+  H₀  |   \ d  |  H₁
+      |     \  |
+      *--------*
+          g(p)
+
+Here, `H₀ = H.to_path x₀` is the path from `f(x₀)` to `g(x₀)`,
+and similarly for `H₁`. Similarly, `f(p)` denotes the
+path in Y that the induced map `f` takes `p`, and similarly for `g(p)`.
+
+Finally, `d`, the diagonal path, is H(0 ⟶ 1, p), the result of the induced `H` on
+`path.homotopic.prod (0 ⟶ 1) p`, where `(0 ⟶ 1)` denotes the path from `0` to `1` in `I`.
+
+It is clear that the diagram commutes (`H₀ ≫ g(p) = d = f(p) ≫ H₁`), but unfortunately,
+many of the paths do not have defeq starting/ending points, so we end up needing some casting.
+-/
+
+/-- Interpret a homotopy `H : C(I × X, Y) as a map C(ulift I × X, Y) -/
+def ulift_map : C(Top.of (ulift.{u} I × X), Y) :=
+⟨λ x, H (x.1.down, x.2),
+  H.continuous.comp ((continuous_induced_dom.comp continuous_fst).prod_mk continuous_snd)⟩
+
+@[simp] lemma ulift_apply (i : ulift.{u} I) (x : X) : H.ulift_map (i, x) = H (i.down, x) := rfl
+
+/-- An abbreviation for `prod_to_prod_Top`, with some types already in place to help the
+ typechecker. In particular, the first path should be on the ulifted unit interval. -/
+abbreviation prod_to_prod_Top_I {a₁ a₂ : Top.of (ulift I)} {b₁ b₂ : X}
+  (p₁ : from_top a₁ ⟶ a₂) (p₂ : from_top b₁ ⟶ b₂) :=
+@category_theory.functor.map _ _ _ _ (prod_to_prod_Top (Top.of $ ulift I) X)
+  (a₁, b₁) (a₂, b₂) (p₁, p₂)
+
+/-- The diagonal path `d` of a homotopy `H` on a path `p` -/
+def diagonal_path : from_top (H (0, x₀)) ⟶ H (1, x₁) :=
+(πₘ H.ulift_map).map (prod_to_prod_Top_I uhpath01 p)
+
+/-- The diagonal path, but starting from `f x₀` and going to `g x₁` -/
+def diagonal_path' : from_top (f x₀) ⟶ g x₁ :=
+hcast (H.apply_zero x₀).symm ≫ (H.diagonal_path p) ≫ hcast (H.apply_one x₁)
+
+/-- Proof that `f(p) = H(0 ⟶ 0, p)`, with the appropriate casts -/
+lemma apply_zero_path : (πₘ f).map p = hcast (H.apply_zero x₀).symm
+  ≫ (πₘ H.ulift_map).map (prod_to_prod_Top_I (𝟙 (ulift.up 0)) p)
+  ≫ hcast (H.apply_zero x₁) :=
```

Comment:
Usually when line breaking near an operator, the operator goes at the end of the line, rather than the beginning.